src/main/java/org/github/guppy/backend/models/Product.java

```diff
@@ -0,0 +1,39 @@
+package org.github.guppy.backend.models;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import java.util.UUID;
+
+@Getter
+@Setter
+@Entity
+public class Product {
+
+    @Id
+    @GeneratedValue
+    private UUID id;
```

Comment:
MySQL does not have a UUID type. The side effect of this is the column ID for the Product table is created as a BINARY(16) column.

**What the hell does that mean?**
Thanks to the god of software, Java, and Hibernate, with this new code, as soon as our application starts, a table named `Product` will be automatically created in our DB. Check the `applications.properties` file where you are also adding `spring.jpa.hibernate.ddl-auto=create`. This line tells Spring to automatically create a DB based on the models with the `@Entity` annotation.
When creating a new row (mean adding new products) in the DB, the ID value for each product will be stored as a BINARY(16) value in the DB.

**I still have no fucking idea what you are talking about!**
UUID stands for **Universally unique identifier** -> https://en.wikipedia.org/wiki/Universally_unique_identifier
It's a common thing to use in software to identify different objects across our organization. Each Product, in our case, will be uniquely identified by one of these.

**OK, so what's wrong with my code?**
Nothing, but if you look at a UUID as a human, it looks like this `0a693771-1a86-44a4-ac68-9b7a54847d5a`. (more examples here -> https://www.uuidgenerator.net/).
In case something goes wrong, I can send a message to my co-worker saying _"hey, the product `0a693771-1a86-44a4-ac68-9b7a54847d5a` appears to have a wrong image URL stored with it"._
_"Cool, let's look at that Product in our table to see if there is a value for an image URL on that UUID"_
This is how the products table will look to us humans that usually don't understand BINARY:
<img width="346" alt="image" src="https://user-images.githubusercontent.com/31945585/189550279-9001e51a-e8de-48ed-91c3-83c73a7a8a99.png">
On a production-ready environment (means a real application), this is sometimes desired and normal to see. In our learning case, this will only difficult things for us if we want to navigate the DB on our own.

**Impressive! How can I make this stupid column look like something we humans can read?**
There is a hacky way of doing this, by forcing the column to be of type `VARCHAR` which for us humans means, use letters and numbers, please.
You can add these two extra annotations to force the ID column to be human readable:
```
    @Id
    @GeneratedValue
    @Column(name="id", columnDefinition = "VARCHAR(36)")
    @Type(type = "uuid-char")
    private UUID id;
```
This will force the column to be of type VARCHAR, making our products column to look like:
<img width="346" alt="image" src="https://user-images.githubusercontent.com/31945585/189550879-2402645a-d3ba-4a84-b7cd-c7555afc4126.png">


Other DB systems, like Postgress, have UUID as a column type, and probably this hacky way is not needed.

Reply:
A pleno Juli, totalmente valido el comentario! Cuando puedas @pedro9747 aplica el cambio!

Reply:
Thanks for all your wisdom hope some day I can achieve your knowledge and have the power to stop a PR